manager/dispatcher/dispatcher_test.go

```diff
@@ -543,14 +568,42 @@ func TestAssignmentsAddingTasks(t *testing.T) {
 	assert.Empty(t, resp.Changes)
 
 	// create the relevant secrets, configs, and tasks and update the tasks
-	secrets, configs, tasks := makeTasksAndDependencies(t, nodeID)
+	secrets, configs, resourceRefs, tasks := makeTasksAndDependencies(t, nodeID)
 	err = gd.Store.Update(func(tx store.Tx) error {
 		for _, secret := range secrets[:len(secrets)-1] {
 			assert.NoError(t, store.CreateSecret(tx, secret))
 		}
 		for _, config := range configs[:len(configs)-1] {
 			assert.NoError(t, store.CreateConfig(tx, config))
 		}
+		// make dummy secrets and configs for resourceRefs
+		for _, resourceRef := range resourceRefs {
+			switch resourceRef.ResourceType {
+			case api.ResourceType_SECRET:
+				dummySecret := &api.Secret{
+					ID: resourceRef.ResourceID,
+					Spec: api.SecretSpec{
+						Annotations: api.Annotations{
+							Name: fmt.Sprintf("dummy_secret_%s", resourceRef.ResourceID),
+						},
+						Data: []byte(fmt.Sprintf("secret_%s", resourceRef.ResourceID)),
+					},
+				}
+				assert.NoError(t, store.CreateSecret(tx, dummySecret))
+			case api.ResourceType_CONFIG:
+				dummyConfig := &api.Config{
+					ID: resourceRef.ResourceID,
+					Spec: api.ConfigSpec{
+						Annotations: api.Annotations{
+							Name: fmt.Sprintf("dummy_config_%s", resourceRef.ResourceID),
+						},
+						Data: []byte(fmt.Sprintf("config_%s", resourceRef.ResourceID)),
+					},
+				}
+				assert.NoError(t, store.CreateConfig(tx, dummyConfig))
+			}
+		}
```

Comment:
This looks like it's duplicated a few times. It would be great to deduplicate it.